Add test for extended proofreading.

megangardner opened the pull request:
#### 513756b52c10e5b18306f758ef0038ff5918ce91
<pre>
Add test for extended proofreading.
<a href="https://bugs.webkit.org/show_bug.cgi?id=303967">https://bugs.webkit.org/show_bug.cgi?id=303967</a>
<a href="https://rdar.apple.com/165892721">rdar://165892721</a>

Reviewed by Wenson Hsieh and Aditya Keerthi.

Adding a test. This didn&apos;t work on iOS before because
strings are tokenized differently on mac and iOS and it had a newline in the key.

* LayoutTests/editing/input/cocoa/extended-proofreading-expected.txt: Added.
* LayoutTests/editing/input/cocoa/extended-proofreading.html: Added.
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/304303@main">https://commits.webkit.org/304303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/566ec193b66be7b1d2efe1f38540069ed99bfd4d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135108 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7500 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46364 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142614 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86916 "Built successfully") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8135 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7349 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103241 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70482 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138054 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5783 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121091 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84097 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5589 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3200 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3208 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114801 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39261 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145326 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7184 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39834 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111625 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7228 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6021 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111992 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28429 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5421 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117385 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61114 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7237 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35538 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6998 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7223 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7101 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/editing/input/cocoa/extended-proofreading.html:
```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html><!-- webkit-test-runner [ spellCheckingDots=true extendedProofreadingEnabled=true ] -->
```
```suggestion
<!DOCTYPE html><!-- webkit-test-runner [ spellCheckingDots=true ExtendedProofreadingEnabled=true ] -->
```

`spellCheckingDots` is lowercase because it's from `TestOptions`. `ExtendedProofreadingEnabled` is uppercase because that's the name in `UnifiedWebPreferences`.

megangardner replied:
Ahh yes, this caused me so much trouble previously.

hortont424 replied:
(It's a bit annoying that unknown names don't throw an error, and that they're case sensitive at all...)

pxlcoder commented on LayoutTests/editing/input/cocoa/extended-proofreading.html:
```diff
@@ -0,0 +1,70 @@
+
+if (window.internals) {
+    internals.setContinuousSpellCheckingEnabled(true);
+    internals.settings.setExtendedProofreadingEnabled(true);
```
This shouldn't be needed since you enable in the test header.

pxlcoder replied:
Since it looks like I'm referring to the entire code block in the GitHub UI... I mean specifically `internals.settings.setExtendedProofreadingEnabled(true);`.

pxlcoder commented on LayoutTests/editing/input/cocoa/extended-proofreading.html:
```diff
@@ -0,0 +1,70 @@
+<meta charset="UTF-8">
+<style>
+div {
+  font-size: 300%;
```
```suggestion
    font-size: 300%;
```

whsieh commented on LayoutTests/editing/input/cocoa/extended-proofreading.html:
```diff
@@ -0,0 +1,70 @@
+    if (window.testRunner) {
+        shouldBecomeEqual('internals.hasGrammarMarker(15, 2)', 'true', function() {
+            finishJSTest();
+        });
```
```suggestion
        await shouldBecomeEqual('internals.hasGrammarMarker(15, 2)', 'true');
        finishJSTest();
```

megangardner committed:
Add test for extended proofreading.
https://bugs.webkit.org/show_bug.cgi?id=303967
rdar://165892721

Reviewed by Wenson Hsieh and Aditya Keerthi.

Adding a test. This didn't work on iOS before because
strings are tokenized differently on mac and iOS and it had a newline in the key.

* LayoutTests/editing/input/cocoa/extended-proofreading-expected.txt: Added.
* LayoutTests/editing/input/cocoa/extended-proofreading.html: Added.
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/304303@main
